src/common.c

```diff
@@ -1129,10 +1141,24 @@ long get_system_cpus(void) {
 
     debug(D_SYSTEM, "System has %d processors.", processors);
     return processors;
+
+    #endif /* __APPLE__ */
 }
 
 pid_t pid_max = 32768;
 pid_t get_system_pid_max(void) {
+    #ifdef __APPLE__
+        int32_t tmp_pid_max;
+
+        if (unlikely(GETSYSCTL("kern.maxproc", tmp_pid_max))) {
```

Comment:
You are absolutely right. This was my fault. I found this:
http://apple.stackexchange.com/questions/51119/whats-the-maximum-pid-for-mac-os-x

On my device I found no PID higher than 99999. The current default is 32768.

Currently the highest PID on my running macOS is 95999. Maybe we should set 99999 as the default for macOS.
@ktsaou Should I set it also as default if not found on another os?